kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkScanBuilder.java

```diff
@@ -75,16 +84,63 @@ public void pruneColumns(StructType requiredSchema) {
                 .toArray(StructField[]::new));
   }
 
+  @Override
+  public Filter[] pushFilters(Filter[] filters) {
+    List<Filter> kernelSupportedFilters = new ArrayList<>();
+    List<Predicate> convertedKernelPredicates = new ArrayList<>();
+    List<Filter> dataFilterList = new ArrayList<>();
+    List<Filter> postScanFilters = new ArrayList<>();
+
+    for (Filter filter : filters) {
+      ExpressionUtils.FilterClassificationResult classification =
+          ExpressionUtils.classifyFilter(filter, partitionColumnSet);
+      // Collect kernel predicates if supported
+      if (classification.isKernelSupported) {
+        convertedKernelPredicates.add(classification.kernelPredicate.get());
+
+        if (!classification.isPartialConversion) {
+          // Add filter to kernelSupportedFilters if it is fully converted
+          // TODO: add partially converted Spark filter as well
+          // right now we only have the partially converted kernel predicate
+          kernelSupportedFilters.add(filter);
+        }
+      }
+      // Collect data filters
+      if (classification.isDataFilter) {
+        dataFilterList.add(filter);
+      }
+      // Collect post-scan filters
```

Comment:
nit: maybe add a sentence to say why we need to add to post scan filters on these three condition. Not obvious for other reader without this context